### PR TITLE
Name fasttrack questions with the new suggested scheme

### PR DESCRIPTION
**Pull request by chrisjpurdy**
Also makes sure that gameboard representation of fasttrack question titles is the same as their title on a question page (i.e. has the i for upper or ii for lower)

image
The bottom two questions are `ft_top_ten` (I think), and the others are `ft_upper` and `ft_lower`, distinguished by `i` or `ii` respectively.

See [this card](https://trello.com/c/2sahDjkq/2130-change-to-automatic-naming-of-fast-track-maths-questions) for details.

### CODE REVIEW

**Review by mlt47 — CHANGES_REQUESTED** [review]
Small refactoring suggestion.
Moving that pure utility function out into a service will also hopefully reduce the chance of us inadvertently crating a cyclic dependency in the imports!

We also display the title in other places such as search results, and navigation links as you move between questions in a gameboard so worth having a little search for those.

File: src/app/components/pages/Gameboard.tsx [hunk 1]
```diff
@@ -41,6 +42,18 @@ function getTags(docTags?: string[]) {
...
 
+function generateQuestionTitle(question : GameboardItem) {
+    if (question.title) {
+        // FastTrack title renaming - here for consistency with how questions are named on FastTrack boards
+        if (question.tags?.includes('ft_upper') || question.tags?.includes('ft_lower')) {
+            return question.title + " " + fastTrackConceptEnumerator(question.id as string) + (question.tags.includes('ft_lower') ? "ii" : "i");
+        } else {
+            return question.title;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (chrisjpurdy) [reply src/app/components/pages/Gameboard.tsx]: Done!

File: src/app/components/pages/Question.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ interface QuestionPageProps {
     location: {search: string};
 }
 
-function fastTrackConceptEnumerator(questionId: string) {
+export function fastTrackConceptEnumerator(questionId: string) {
     // Magic, unfortunately
     return "_abcdefghijk".indexOf(questionId.split('_')[2].slice(-1));
 }
```
[the remark this answers is not in the document]
Comment (chrisjpurdy) [reply src/app/components/pages/Question.tsx]: Done this as well - I've tried to make sure that question titles are consistent everywhere, but in navigation links there isn't enough information about the content being linked to yet to call `generateQuestionTitle` (missing `type` and `tags`)

**Review by mlt47 — APPROVED** [review]
Great

**Pull request merged**